### PR TITLE
List of speakers with angularjs

### PR DESCRIPTION
**Pull request by emanuelschuetze**
- Add/remove user to/from list
- Open/close list

### CODE REVIEW

File: openslides/agenda/static/js/agenda/agenda.js [hunk 1]
```diff
@@ -115,9 +115,32 @@ angular.module('OpenSlidesApp.agenda.site', ['OpenSlidesApp.agenda'])
...
 })
 
-.controller('ItemDetailCtrl', function($scope, Agenda, User, item) {
+.controller('ItemDetailCtrl', function($scope, $http, Agenda, User, item) {
     Agenda.bindOne(item.id, $scope, 'item');
     User.bindAll({}, $scope, 'users');
+    $scope.speaker = {};
+
+    // close/open list of speakers
+    $scope.closeList = function (item_id, bool) {
```
Comment (ostcar) [inline openslides/agenda/static/js/agenda/agenda.js]: you do not use 'item_id'.

Could you rename 'bool' to a name that says what the value could be, and not which type it is?

File: openslides/agenda/static/js/agenda/agenda.js [hunk 2]
```diff
@@ -115,9 +115,32 @@ angular.module('OpenSlidesApp.agenda.site', ['OpenSlidesApp.agenda'])
...
+        item.speaker_list_closed = bool;
+        Agenda.save(item);
+    };
+    // add user to list of speakers
+    $scope.addSpeaker = function (item_id, user_id) {
```
Comment (ostcar) [inline openslides/agenda/static/js/agenda/agenda.js]: proposal: rename item_id to itemId and user_id to userId

File: openslides/agenda/static/js/agenda/agenda.js [hunk 3]
```diff
@@ -115,9 +115,32 @@ angular.module('OpenSlidesApp.agenda.site', ['OpenSlidesApp.agenda'])
...
+        $http.post('/rest/agenda/item/'+item_id+'/manage_speaker/', {'user': user_id})
```
Comment (ostcar) [inline openslides/agenda/static/js/agenda/agenda.js]: proposal: Add spaces before and after the + sign

File: openslides/agenda/static/js/agenda/agenda.js [hunk 4]
```diff
@@ -115,9 +115,32 @@ angular.module('OpenSlidesApp.agenda.site', ['OpenSlidesApp.agenda'])
...
+            .success(function(data){
+                $scope.alert.show = false;
+            })
+            .error(function(data){
+                $scope.alert = { type: 'danger', msg: data.detail, show: true };
+            });
+    };
+    // delete speaker(!) from list of speakers
+    $scope.removeSpeaker = function (item_id, speaker_id) {
+        $http.delete('/rest/agenda/item/'+item_id+'/manage_speaker/',
```
Comment (ostcar) [inline openslides/agenda/static/js/agenda/agenda.js]: Here the same

File: openslides/agenda/static/js/agenda/agenda.js [hunk 5]
```diff
@@ -115,9 +115,32 @@ angular.module('OpenSlidesApp.agenda.site', ['OpenSlidesApp.agenda'])
...
+                { headers: { 'Content-Type': 'application/json' },
```
Comment (ostcar) [inline openslides/agenda/static/js/agenda/agenda.js]: Proposal: Remove the space after the two opening \{

File: openslides/agenda/static/js/agenda/agenda.js [hunk 6]
```diff
@@ -115,9 +115,32 @@ angular.module('OpenSlidesApp.agenda.site', ['OpenSlidesApp.agenda'])
...
+                  data: JSON.stringify({ speaker: speaker_id })});
```
Comment (ostcar) [inline openslides/agenda/static/js/agenda/agenda.js]: Here the same

File: openslides/agenda/static/templates/agenda/item-detail.html [hunk 7]
```diff
@@ -31,26 +31,53 @@ <h3 os-perm="agenda.can_manage" translate>Comment</h3>
 </div>
 
 
-<h3 translate>List of speakers</h3>
+<h3 translate>List of speakers
```
Comment (ostcar) [inline openslides/agenda/static/templates/agenda/item-detail.html]: The closing \<h3> tag is not optional: http://www.w3.org/TR/html5/syntax.html#optional-tags

Comment (ostcar) [inline openslides/agenda/static/templates/agenda/item-detail.html]: Please put only the headline into the \<h3> tag and the links to open and close the list after it.

File: openslides/agenda/static/js/agenda/agenda.js (hunk 1 above)
Comment (emanuelschuetze) [inline openslides/agenda/static/js/agenda/agenda.js]: done

File: openslides/agenda/static/js/agenda/agenda.js (hunk 2 above)
Comment (emanuelschuetze) [inline openslides/agenda/static/js/agenda/agenda.js]: done

File: openslides/agenda/static/js/agenda/agenda.js (hunk 3 above)
Comment (emanuelschuetze) [inline openslides/agenda/static/js/agenda/agenda.js]: done

File: openslides/agenda/static/js/agenda/agenda.js (hunk 4 above)
Comment (emanuelschuetze) [inline openslides/agenda/static/js/agenda/agenda.js]: done

File: openslides/agenda/static/js/agenda/agenda.js (hunk 5 above)
Comment (emanuelschuetze) [inline openslides/agenda/static/js/agenda/agenda.js]: done

File: openslides/agenda/static/js/agenda/agenda.js (hunk 6 above)
Comment (emanuelschuetze) [inline openslides/agenda/static/js/agenda/agenda.js]: done

File: openslides/agenda/static/templates/agenda/item-detail.html (hunk 7 above)
Comment (emanuelschuetze) [inline openslides/agenda/static/templates/agenda/item-detail.html]: To use label-span-tag within of h3-tag is recommended by bootstrap, see http://getbootstrap.com/components/#example
And I like it to have the button beside the label...

**Comment by emanuelschuetze** [comment]
Done. Ready for next review.

**Comment by normanjaeckel** [comment]
Ok (untested).

File: openslides/agenda/static/templates/agenda/item-detail.html (hunk 7 above)
Comment (ostcar) [inline openslides/agenda/static/templates/agenda/item-detail.html]: ok

**Comment by ostcar** [comment]
ok, please rebase and merge

**Comment by emanuelschuetze** [comment]
Rebased. Ready for merge.

**Pull request merged**